src/components/EditorWidgets/Date/DateControl.js

```diff
@@ -34,16 +34,35 @@ export default class DateControl extends React.Component {
     }
   }
 
+  // Date is valid if datetime is a moment or Date object otherwise it's a string.
+  // Handle the empty case, if the user wants to empty the field.
+  isValid = datetime => (moment.isMoment(datetime) || datetime instanceof Date || datetime === '');
```

Comment:
`isValid` has a specific use for widget controls, which we're not trying to use here, so can we change this to something else, maybe `isValidDate`?